numpy/linalg/tests/test_linalg.py

```diff
@@ -1761,6 +1763,31 @@ def test_xerbla_override():
             raise SkipTest('Numpy xerbla not linked in.')
 
 
+def test_sdot_bug_8577():
+    # Regression test that loading certain other libraries does not
+    # result to wrong results in float32 linear algebra.
+    #
+    # There's a bug gh-8577 on OSX that can trigger this, and perhaps
+    # there are also other situations in which it occurs.
+    #
+    # Do the check in a separate process.
+
+    bad_libs = ['PyQt5.QtWidgets', 'IPython']
+
+    for bad_lib in bad_libs:
+        code = textwrap.dedent("""
+        import sys, numpy as np
+        try:
+            import {bad_lib}
```

Comment:
Worth repeating the test with the order swapped?

Reply:
Not sure, iiuc such failure was not reported.

Reply:
ok, added as it probably doesn't hurt to check for it too